### ISSUE TITLE
Deleted routes hide visible routes

### ISSUE THREAD

**Issue by magiconair** [eBay/fabio#57]
All versions of fabio have the following bug:

If you delete all routes of a service with the `route del service` manual override then paths registered by this service are still considered for routing. 

Given this routing table

```
route add serviceA /foo http://127.0.0.1:5000/
route add serviceB /foo/bar http://127.0.0.1:5001/
route del serviceB
```

the log will print the correct routing table

```
route add serviceA /foo http://127.0.0.1:5000/
```

but internally there is still a route for `/foo/bar` with no targets which is evaluated during lookup. Therefore, a request for `/foo/bar` will not be routed to `serviceA` but to nowhere since the dangling route has no targets.

The current workaround is to add a manual override for the specific route to serviceA, e.g.

```
route add serviceA /foo/bar http://127.0.0.1:5000/
```

**Comment by magiconair** [comment]
Thanks for @ilya-pi for finding it.

**Comment by magiconair** [comment]
Hmm, the fix for this bug fundamentally conflicts with issue #52 since there the deleted routes are only marked but not removed. I may have to rethink that

**Comment by magiconair** [comment]
I'll roll this fix first and then refactor for issue #52.
